code/standards/checkstyle.xml

```diff
@@ -75,13 +83,21 @@
     <module name="FinalClass"/>
     <module name="HideUtilityClassConstructor"/>
     <module name="InterfaceIsType"/>
-    <module name="VisibilityModifier"/>
+    <module name="VisibilityModifier">
+      <property name="protectedAllowed" value="true" />
+      <property name="allowPublicFinalFields" value="true" />
```

Comment:
Why?  I know I have a few classes where I've bent the rules, but that doesn't make them good design :)
the danger with public final is if that item is mutable, OO was destroyed in the process.  Shouldn't this be suppressed on an individual basis when it's reviewed by a person, rather than globally not enforced?